Branch-SDK/src/main/java/io/branch/coroutines/DeviceSignals.kt

```diff
@@ -0,0 +1,52 @@
+package io.branch.coroutines
+
+import android.content.Context
+import android.webkit.WebSettings
+import android.webkit.WebView
+import io.branch.referral.BranchLogger.e
+import io.branch.referral.BranchLogger.v
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+
+
+/**
+ * Returns the user agent string on a background thread via static class WebSettings
+ */
+suspend fun getUserAgentAsync(context: Context): String? {
+    return withContext(Dispatchers.Default) {
+        v("Begin getUserAgentAsync " + Thread.currentThread())
+
+        var result: String? = null
+        try {
+            result = WebSettings.getDefaultUserAgent(context)
+        }
+        catch (exception: Exception) {
+            e("Failed to retrieve userAgent string. " + exception.message)
+        }
+
+        v("End getUserAgentAsync " + Thread.currentThread() + " " + result)
+        result
+    }
+}
+
+/**
+ * Returns the user agent string on the main thread via WebView instance.
+ */
+suspend fun getUserAgentSync(context: Context): String?{
```

Comment:
Now that this isn't on startup, can it be the default option?

Reply:
That would be a behavior change for apps that haven't opted into this api. While it is dispatched, it still requires the main thread for a few hundred ms which backgrounded integrations should be aware of first. Could be in scope for future.